source/common/common/matchers.cc

```diff
@@ -78,6 +78,18 @@ bool MetadataMatcher::match(const envoy::api::v2::core::Metadata& metadata) cons
   case ProtobufWkt::Value::kBoolValue:
     return (bool_matcher_.has_value() && *bool_matcher_ == value.bool_value());
   case ProtobufWkt::Value::kListValue:
+    for (int i = 0; i < value.list_value().values_size(); ++i) {
```

Comment:
I don't think we should automatically extend the value matchers to list (and treat it is matched when one of the list element matches to it), can we make a `ListMatch` in matcher/metadata.proto, and make this behavior explicitly from the API? In future we may want to do more than oneof match for lists.

Reply:
+1